### ISSUE TITLE
Segmentation Fault on Drop

### ISSUE THREAD

**Issue by xuorig** [alexmoon/bluest#7]
Dropping the `Adapter` seems to result in Segmentation Fault. On Macbook Pro with Ventura.

I'm running with the latest `main` which includes this [fix](https://github.com/alexmoon/bluest/pull/1)

Repro:

```rust
#[tokio::main]
async fn main() -> Result<(), Box<dyn Error>> {
    let adapter = Adapter::default().await.ok_or("Bluetooth adapter not found")?;
    adapter.wait_available().await?;
    Ok(())
}
```

**Comment by xuorig** [comment]
```
(lldb) bt
* thread #1, name = 'main', queue = 'com.apple.main-thread', stop reason = EXC_BAD_ACCESS (code=1, address=0xbeaddead80f0)
  * frame #0: 0x000000010017013c wahoo-rs`core::sync::atomic::atomic_load::hae0f581a02711306(dst=0x0000beaddead80f0, order=Acquire) at atomic.rs:3185:24
    frame #1: 0x0000000100173a60 wahoo-rs`std::sys_common::lazy_box::LazyBox$LT$T$GT$::get_pointer::hfb57f462b6961df4 [inlined] core::sync::atomic::AtomicPtr$LT$T$GT$::load::hf04614f1d318cf8f(self=<unavailable>) at atomic.rs:1325:18
    frame #2: 0x0000000100173a4c wahoo-rs`std::sys_common::lazy_box::LazyBox$LT$T$GT$::get_pointer::hfb57f462b6961df4(self=0x0000beaddead80f0) at lazy_box.rs:49:28
    frame #3: 0x000000010015359c wahoo-rs`std::sys::unix::locks::pthread_rwlock::RwLock::write::ha3e78bbced14ac09 [inlined] _$LT$std..sys_common..lazy_box..LazyBox$LT$T$GT$$u20$as$u20$core..ops..deref..Deref$GT$::deref::hfbf191e3f6c9eda8(self=0x0000beaddead80f0) at lazy_box.rs:72:20
    frame #4: 0x0000000100153598 wahoo-rs`std::sys::unix::locks::pthread_rwlock::RwLock::write::ha3e78bbced14ac09(self=0x0000beaddead80f0) at pthread_rwlock.rs:133:21
    frame #5: 0x0000000100151538 wahoo-rs`std::sync::rwlock::RwLock$LT$T$GT$::write::hf3f863ae5cbaf3ad(self=0x0000beaddead80f0) at rwlock.rs:298:13
    frame #6: 0x000000010013390c wahoo-rs`_$LT$async_broadcast..InactiveReceiver$LT$T$GT$$u20$as$u20$core..ops..drop..Drop$GT$::drop::hfe489cbb82f738da(self=0x00006000000080e0) at lib.rs:1872:25
    frame #7: 0x0000000100130910 wahoo-rs`core::ptr::drop_in_place$LT$async_broadcast..InactiveReceiver$LT$bluest..corebluetooth..delegates..CentralEvent$GT$$GT$::ha15a426136d9434d((null)=0x00006000000080e0) at mod.rs:497:1
    frame #8: 0x0000000100130e64 wahoo-rs`core::ptr::drop_in_place$LT$alloc..boxed..Box$LT$async_broadcast..InactiveReceiver$LT$bluest..corebluetooth..delegates..CentralEvent$GT$$GT$$GT$::h74af67b5ae5969c8((null)=0x000000016fdfc118) at mod.rs:497:1
    frame #9: 0x0000000100144cac wahoo-rs`core::mem::drop::h4ab7e0960ed72785(_x=0x00006000000080e0) at mod.rs:987:24
    frame #10: 0x0000000100138c24 wahoo-rs`bluest::corebluetooth::delegates::CentralDelegate::dealloc::h6a1bc17fde8752cf(this=0x0000600000205980, _sel=Sel @ 0x000000016fdfc198) at delegates.rs:304:17
    frame #11: 0x000000010017f198 wahoo-rs`_$LT$objc..rc..strong..StrongPtr$u20$as$u20$core..ops..drop..Drop$GT$::drop::h8c1128fd2e8632ab(self=0x000000016fdfd780) at strong.rs:48:13
    frame #12: 0x0000000100178364 wahoo-rs`core::ptr::drop_in_place$LT$objc..rc..strong..StrongPtr$GT$::hbab83fdfa9fad057((null)=0x000000016fdfd780) at mod.rs:497:1
    frame #13: 0x0000000100013abc wahoo-rs`core::ptr::drop_in_place$LT$objc_id..id..Id$LT$bluest..corebluetooth..delegates..CentralDelegate$C$objc_id..id..Shared$GT$$GT$::h908ffda0bddc227b((null)=0x000000016fdfd780) at mod.rs:497:1
    frame #14: 0x0000000100013f18 wahoo-rs`core::ptr::drop_in_place$LT$bluest..corebluetooth..adapter..AdapterImpl$GT$::h9b41d3a99e1eb4de((null)=0x000000016fdfd770) at mod.rs:497:1
    frame #15: 0x0000000100013dd4 wahoo-rs`core::ptr::drop_in_place$LT$bluest..adapter..Adapter$GT$::hbfeb455000153992((null)=0x000000016fdfd770) at mod.rs:497:1
    frame #16: 0x0000000100012550 wahoo-rs`core::mem::drop::h11db5b8d9ae05de2(_x=<unavailable>) at mod.rs:987:24
    frame #17: 0x0000000100003dfc wahoo-rs`wahoo_rs::main::_$u7b$$u7b$closure$u7d$$u7d$::hb8cf8cdfba8ad9ed((null)=0x000000016fdfdf98) at main.rs:33:5
```

**Comment by alexmoon** [comment]
Thanks for the detailed report. Just a stupid copy/paste bug. Should be fixed now in v0.6.5.
